File: VB/DXM3_HtmlEditor/Controllers/HomeController.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports System.Web.Mvc
Imports DevExpress.Web.Mvc

Namespace DXM3_HtmlEditor.Controllers
	Public Class HomeController
		Inherits Controller
		Public Function Index() As ActionResult
			Return View()
		End Function

		Public Function HtmlEditorPartial() As ActionResult
			Return PartialView()
		End Function

		<HttpPost> _
		Public Function SubmitHtml() As ActionResult
			ViewBag.PostedHtml = HtmlEditorExtension.GetHtml("htmlEditor")

			Return View("Index")
		End Function
	End Class
End Namespace

File: CS/DXM3_HtmlEditor/Controllers/HomeController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;
using DevExpress.Web.Mvc;

namespace DXM3_HtmlEditor.Controllers {
    public class HomeController : Controller {
        public ActionResult Index() {
            return View();
        }

        public ActionResult HtmlEditorPartial() {
            return PartialView();
        }

        [HttpPost]
        public ActionResult SubmitHtml() {
            ViewBag.PostedHtml = HtmlEditorExtension.GetHtml("htmlEditor");

            return View("Index");
        }
    }
}
